pkg/daemon/controller/controller.go

```diff
@@ -0,0 +1,324 @@
+			for _, addr := range node.Status.Addresses {
+				if addr.Address == localAddr.String() {
+					return &node, nil
+				}
+			}
+		}
+	}
+	return nil, errors.New("unable to find associated node")
+}
+
+// NewServiceController returns a pointer to a ServiceController object
+func NewServiceController(client client.Client, mgr winsvc.Mgr, nodeName string) *ServiceController {
+	return &ServiceController{client: client, svcMgr: mgr, ctx: context.TODO(), nodeName: nodeName}
+}
+
+// existingServices returns a map with the keys being all Windows services currently present on the VM
+func (c *ServiceController) existingServices() (map[string]struct{}, error) {
+	// The most reliable way to determine if a service exists or not is to do a 'list' API call. It is possible to
+	// remove this call, and parse the error messages of a service 'open' API call, but I find that relying on human
+	// readable errors could cause issues when providing compatibility across different versions of Windows.
+	svcList, err := c.svcMgr.ListServices()
+	if err != nil {
+		return nil, err
+	}
+	svcs := make(map[string]struct{})
+	for _, service := range svcList {
+		svcs[service] = struct{}{}
+	}
+	return svcs, nil
+}
+
+// reconcileServices ensures that all the services passed in via the services slice are created, configured properly
+// and started
+func (c *ServiceController) reconcileServices(services []servicescm.Service) error {
+	// go through svcList
+	// sort services in ascending priority
+	sort.Slice(services, func(i, j int) bool {
+		return services[i].Priority < services[j].Priority
+	})
+	existingSvcs, err := c.existingServices()
+	if err != nil {
+		return errors.Wrap(err, "could not determine existing Windows services")
+	}
+	for _, service := range services {
+		var winSvcObj winsvc.Service
+		if _, present := existingSvcs[service.Name]; !present {
+			// create a service placeholder
+			winSvcObj, err = c.svcMgr.CreateService(service.Name, "", mgr.Config{})
+			if err != nil {
+				return nil
+			}
+			klog.Infof("created service %s", service.Name)
+		} else {
+			// stop and open the service
+			winSvcObj, err = c.svcMgr.OpenService(service.Name)
+			if err != nil {
+				return nil
+			}
+		}
+		if err := c.reconcileService(winSvcObj, service); err != nil {
+			return err
+		}
+		klog.Infof("successfully reconciled service %s", service)
+	}
+	return nil
+}
+
+// reconcileService ensures the given service is running and configured according to the expected definition given
+func (c *ServiceController) reconcileService(service winsvc.Service, expected servicescm.Service) error {
+	config, err := service.Config()
+	if err != nil {
+		return err
+	}
+	cmd, err := c.expectedServiceCommand(expected)
+	if err != nil {
+		return err
+	}
+
+	updateRequired := false
+	if config.BinaryPathName != cmd {
+		config.BinaryPathName = cmd
+		updateRequired = true
+	}
+
+	expectedDescription := fmt.Sprintf("OpenShift %s", expected.Name)
```

Comment:
Shouldn't this use the [ManagedTag constant](https://github.com/sebsoto/windows-machine-config-operator/blob/daemonController/pkg/windows/windows.go#L97)?